download_action.go

```diff
@@ -11,10 +11,13 @@ import (
 )
 
 type DownloadAction struct {
-	BaseAction `yaml:",inline"`
-	Url        string // URL for downloading
-	Filename   string // File name, overrides the name from URL.
-	Forced     bool   `yaml:"force"` // Forced re-download of URL
+	BaseAction  `yaml:",inline"`
+	Url         string // URL for downloading
+	Filename    string // File name, overrides the name from URL.
+	Forced      bool   `yaml:"force"` // Forced re-download of URL
+	Unpack      bool   // Unpack downloaded file to directory dedicated for download
+	Compression string // compression type
+	Makedir     bool   // Create a dedicated directory for extracting contents
```

Comment:
As discussed on chat, we should always make dedicated directories for extracting so there isn't a chance for accidental overrides. If the user wants download B to override parts of A that should be ensure when deploying the contents onto the filesystem rather then at download time